src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -86,8 +86,28 @@ public async Task<(JObject containerObject, ArraySegment<string> remaining)> Res
             if (methodInfo == null)
                 return (null, null);
 
+            string[] parts;
+            if (includeFullName)
+            {
+                string fullName = methodInfo.IsAsync == 0 ? methodInfo.TypeInfo?.FullName : StripAsyncPartOfFullName(methodInfo.TypeInfo?.FullName);
+                string[] fullNameParts = fullName.Split(".", StringSplitOptions.TrimEntries);
+                int overlappingPartIdx = Enumerable.Range(0, fullNameParts.Length).LastOrDefault(i => fullNameParts[i] == expressionParts[0]);
```

Comment:
`LastOrDefault` here will be using only the parts that matched, not all. Is that the intention?

Reply:
Yes, we need here the last part of full name that is the same as the root of the user's expression. Let's assume the expression is: `c.b.a` and we are checking if it has a part overlapping with full name: `e.d.c.b`. The expected result would be index of "c" in the full name, so that we can take everything that is before "c": `e.d` and prefix it to the user's expresssion: `e.d.c.b.a`. So yes, we are interested only in elements that are equal to "c" in this case.

Reply:
Change into looping.